assignment3.txt

```diff
@@ -0,0 +1,7 @@
+1. it is invalid because a for loop has 3 requirements inside the for loop. (initialisation; condition; incrementat). It does not meet these requirements.
+2. It is an infinite loop. The number will keep getting smaller.
+3. 
+i. n^2
+ii. cube if the if statement is true if not it is only squared.
```

Comment:
The answer is actually O(n^4) - again, for loops that run from 0 to n have the time complexity O(n). So the top loop is O(n), the middle Loop is O(n^2) iterations because it iterates from 1 to i^2. The if statement is O(1 / n) times because of the modulo function. Lastly, the inner loop  is O(n^2) iterations because it iterates from 0 to j and j = i^2 = n^2. 

Therefore, the sum = -(1/12)*n + (3/8)*n^2 - (5/12)*n^3 + (1/8)*n^4  = O(n^4)